librz/util/file.c

```diff
@@ -1255,7 +1255,17 @@ RZ_API RzList *rz_file_globsearch(const char *_globbed_path, int maxdepth) {
 	return files;
 }
 
-RZ_API char *rz_file_path_join(const char *s1, const char *s2) {
+/**
+ * \brief Concatenate two paths to create a new one with s1+s2 with the correct path separator
+ *
+ * \param s1 First path
+ * \param s2 Second path, can be NULL
+ * \return Full path
+ */
+RZ_API RZ_OWN char *rz_file_path_join(const char *s1, const char *s2) {
```

Comment:
Add `RZ_NONNULL` for `s1` and `RZ_NULLABLE` for `s2`, also `rz_return_val_if_fail(s1, NULL)`